cmd/shell/cmd/client.go

```diff
@@ -11,6 +19,22 @@ const (
 )
 
 func init() {
+	client.RegisterFlagBinder(func(fs *pflag.FlagSet, namer naming.Namer) {
+		fs.StringVar(&Auth.Username, namer("auth-username"), Auth.Username, "username for authentication")
+		fs.StringVar(&Auth.Password, namer("auth-password"), Auth.Password, "password for authentication")
+	})
+
+	client.RegisterPreDialer(func(_ context.Context, opts *[]grpc.DialOption) error {
+		cred := Auth
+
+		authHeader := fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(cred.Username+":"+cred.Password)))
```

Comment:
How about when gRPC doesn't have password?

Reply:
Please test it, with or without auth.

Reply:
it works fine.